backend/migration_tools/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,367 @@
+import json
+import logging
+import os
+import time
+from typing import Any, Optional
+
+import psycopg2
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from migration_tools.unstract_migrations import UnstractMigration
+from psycopg2.extensions import connection, cursor
+
+logger = logging.getLogger(__name__)
+
+
+class DataMigrator:
+    def __init__(self, src_db_config, dest_db_config, v2_schema, batch_size=1000):
+        self.src_db_config = src_db_config
+        self.dest_db_config = dest_db_config
+        self.batch_size = batch_size
+        self.v2_schema = v2_schema
+        self.migration_tracking_table = f"{self.v2_schema}.migration_tracking"
+
+    def connect_db(self, db_config) -> connection:
```

Comment:
NIT: @muhammad-ali-e try using a context manager to not worry about closing the connection